### DISCUSSION TITLE
Optimization Discussion

### DISCUSSION THREAD

**Discussion by ilikenwf** [hlky/stable-diffusion-webui#732] [Feature suggestions]
I've seen some interesting modifications people have made, some just to attention.py - cleaning up after stuff is used.

A good discussion ongoing here:
https://gist.github.com/mchaker/e0cf180db6849fa03696293b3fc8f485

Could any/all of these be adapted for this repo? For the attention mods that slow things down, I think that should perhaps require a checkmark or config option, or automatic memory threshold determination - that is, if you don't have a 24GB VRAM card and instead have 11-12 or less, the slow but stable stepping option should kick in for images greater than 768 or a bit more pixels on either dimension....

I think we may already somewhat do this?
https://github.com/CompVis/stable-diffusion/pull/142

Then, more interesting - Doggettx memory optimization: allows rendering 1024x1024 images and even up to 4096x1024:
https://github.com/CompVis/stable-diffusion/compare/main...Doggettx:stable-diffusion:main

Another currently incompatible (or similar vein?) change:
https://github.com/basujindal/stable-diffusion/pull/122

**Comment by extesy** [comment]
https://github.com/CompVis/stable-diffusion/pull/177 is very small but still quite effective.

**Comment by TingTingin** [comment]
I think someone added this but it got lost in the merge

**Reply by ilikenwf** [reply]
Which one?

**Reply by TingTingin** [reply]
The doggetx memory optimizations someone had made a pull request adding it but in the big repo merge it seems it never got added dont know if hlky is gonna add it later

**Reply by TingTingin** [reply]
I was able to find the guy that did it and go straight to the repo i guess people can copy from here https://github.com/swfsql/stable-diffusion/commit/9c9c1ea9dff71582eb207fd3980cc206d7e79d21
just tested and it works!

**Reply by mchaker** [reply]
@TingTingin how does the `swfsql` modification differ from the `Doggettx` modification? I can do a diff but do you know a human-language summary of the difference? 😅 

On first glance, it looks like the `Doggettx` optimizations are more thorough

**Comment by wcarletsdrive** [comment]
I love the upgrades you put on this SD! I wanted to know if anyone could increase the optimization to allow 12gb users to use 1024 x 1024 or 1280 x 768 like Disco Diffusion. 1280 X 768 or 1280 x 1024 would be incredible!! It adds time but this is still better than Disco Diffusion. So it would be an awesome upgrade to add ASAP.

**Comment by mchaker** [comment]
I'm collecting a list of useful optimizations at https://links.sd

**Reply by wcarletsdrive** [reply]
Thanks!!
